train.py

```diff
@@ -10,16 +10,21 @@
 # The hyperparameters
 alpha = 0.1
 gamma = 0.6
-epsilon = 0.1
+epsilon = 0.95
+decay_factor = 0.97
 
 NUM_EPISODES = 100000
 
 
 def update(q_table, env, state):
+    global epsilon
+    
     if random.uniform(0, 1) < epsilon:
         action = env.action_space.sample()
     else:
         action = select_optimal_action(q_table, state, env.action_space)
+        
+    epsilon = min(0.1, epsilon*decay_factor)
```

Comment:
I guess I put it wrong in the issue description (sorry about that), but this should actually be `max`.